packages/runtime/test/execute/expression.test.ts

```diff
@@ -139,17 +140,20 @@ test.serial('async operations run in series', async (t) => {
   t.is(result.data.x, 12);
 });
 
-test.serial('jobs can return undefined', async (t) => {
-  // @ts-ignore violating the operation contract here
-  const job = [() => undefined] as Operation[];
+test.serial(
+  'jobs return null-state instead of undefined or null',
```

Comment:
So currently, this PR makes sure that, `null` or `undefined` aren't possible values returned by an operation. because they get replaced with a `null-state`.

When an operation is executed, 
1. we don't let null or undefined to be passed to the next operation. 
2. we replace that with a marked empty object. 
3. the next operation check whether its input state is marked and get's to know that the previous state actually didn't return state.